### PR TITLE
Hw3_t04

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

**Comment by mirmozavr** [comment]
Need assistance with module imports in a right way with sys.path appends or__init__.py files

**Comment by Cjkjvfnby** [comment]
> Need assistance with module imports in a right way with sys.path appends or__init__.py files

It is a quite vague request.  And you will get a vague answer:  import them in the right way.

To get a more precise answer please add more info to the question:  What do you want to achieve? What is blocking you?  What have you tried so far?

**Comment by mirmozavr** [comment]
In this homework i decided to separate all test files into dedicated test folder. To fix imports for github i tried appending current file directory to sys.path, than addint __init__.py to folders.

**Comment by Cjkjvfnby** [comment]
> In this homework i decided to separate all test files into dedicated test folder. 

I am not sure that this change is present in the branch.

> To fix imports for github 

You can run `pytest` locally in the root directory of the project.  It should produce the same result as on GitHub. 

Put `__init__.py` to folders with your homework, and don't put them to folders with tests.

PS. Please press the preview button before posting the comments. Use ticks around special names like `__inti__.py` (tick -> ` )

**Comment by mirmozavr** [comment]
> I am not sure that this change is present in the branch.
The tree looks like this.
├───hw3
│   ├───tests
│   │   └───test_files.py
│   └───task_files.py

All imports had worked and tests had passed on Github

File: hw3/hw3_t04.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+### Example function signature and call
+"""
+from functools import reduce
+
+
+def is_armstrong(number: int) -> bool:
+    power = len(str(number))
+    return number == reduce(
```
Comment (Cjkjvfnby) [inline hw3/hw3_t04.py]: Do you know why reduce was hidden?  Because it is overused.  You can use another built-in function.

File: hw3/hw3_t04.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        lambda left, right: left + right,
+        map(lambda symbol: int(symbol) ** power, str(number)),
```
Comment (Cjkjvfnby) [inline hw3/hw3_t04.py]: You can replace lamba + map with a list comprehension.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

**Comment by Cjkjvfnby** [comment]
> All imports had worked and tests had passed on Github

I don't think that you need any assistance with it.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**